include/zephyr/bluetooth/mesh/access.h

```diff
@@ -904,18 +916,16 @@ struct bt_mesh_model {
 	/** Model callback structure. */
 	const struct bt_mesh_model_cb * const cb;
 
-#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
-	/* Pointer to the next model in a model extension list. */
-	struct bt_mesh_model *next;
-#endif
-
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
 	/* Pointer to the array of model metadata entries. */
-	struct bt_mesh_models_metadata_entry **metadata;
+	struct bt_mesh_models_metadata_entry ** const metadata;
 #endif
 
+	/** Model ctx variables. */
+	struct bt_mesh_model_ctx * const ctx;
```

Comment:
If you move these three to `bt_mesh_model_ctx`, why not to move `keys`, `keys_cnt`, `groups`, etc. also there? Then this is just going to make code even more complex and unreadable. 

Thus, i don't see any specific need for this structure. I think keeping `elem_idx`, `mod_ix`, `flags` as they were now is fine, and this new structure should not be introduced.

Reply:
The struct `bt_mesh_model_ctx` will make no-const, and `struct bt_mesh_model` will be const.